pages/voting-guides/index.tsx

```diff
@@ -0,0 +1,149 @@
+        result.push(electionInfo);
+      } else {
+        result[i]?.guides.push(value);
+      }
+      return result;
+    }, [] as VotingGuideElection[]);
+  } else return [] as VotingGuideElection[];
+};
+
+const VotingGuides: NextPage<{ mobileNavTitle?: string }> = ({mobileNavTitle}) => {
+
+  const user = useAuth({redirectTo:"/login"}).user; //user will not be null under this line, due to the conditional redirect
+  const votingGuidesByUserId = useVotingGuidesByUserIdQuery({userId:user?.id || ""}); 
+  const [elections, setElections] = useState<VotingGuideElection[]>([]);
+
+  useEffect(()=>{
+    const {data, isError, error} = votingGuidesByUserId;
+  if (isError){
+    alert("Error getting Voting Guides");
+    console.error("Error getting Voting Guides", error);
+  }
+  else{
+    if (data && data.votingGuidesByUserId.length) {
+      const newVotingGuides = data.votingGuidesByUserId as VotingGuideResult[];
+      const elections = getElectionsFromGuides(newVotingGuides);
+      console.log('yes data', elections);
+      setElections(elections);
+    } else {
+      votingGuidesByUserId.refetch().then(res=>{
+        const elections = getElectionsFromGuides(res.data?.votingGuidesByUserId as VotingGuideResult[]);
+        console.log('no data', elections);
+        setElections(elections);
+      });
+    }
+  };
+  },[]);
+
+  const upsertVotingGuide = useUpsertVotingGuideMutation();
+
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  return (
+    <>
+      <Head>
+        <title>Populist - Voting Guides</title>
+        <meta name="description" content="View Voting Guides." />
+      </Head>
+
+      <Layout mobileNavTitle={`${mobileNavTitle || 'Voting Guides'}`} showNavLogoOnMobile={false}>
+        <div className={styles.votingContainer}> 
+          <FlagSection title="Voting Guides">
+            {
+              elections.map(election=>(
+                <div key={election.id}>
+                  <h1 className={styles.votingHeader}>{election.date}</h1>
+                  <h2>{election.title}</h2>
+                  <p className={styles.description}>{election.description}</p>
+                    {
+                      election.guides?.map(guide=><VotingGuideCard guide={guide} key={guide.id} />)
+                    }
+                </div>
+              ))
+            }
+          </FlagSection>
+        </div>
+
+        <div className={styles.roundedCard}>
+          <h2>Add a test guide </h2>
+          <h3>(for CO Primary Election)</h3>
+          <input className={styles.guideInput} type="text" placeholder="Title" onChange={e=>setTitle(e.target.value)} value={title} />
+          <input className={styles.guideInput} type="text" placeholder="Description" onChange={e=>setDescription(e.target.value)} value={description} />
+          <Button large theme="blue" disabled={!title || !description} label="add_guide"
+            onClick={()=>
```

Comment:
Is this temporary? If not, let's get this in a named function instead of defining it inline.

Reply:
agree with @neko-mamushi on this

Reply:
This section is completely temporary which is why I didn't spend much care on it. We just want to have a way to add guides for now to see how it will display in this section. Agree we need an input component but I did this one dirty since it will be removed as soon as we can create voting guides elsewhere